db/migrate/20211115012130_add_annotation_model.rb

```diff
@@ -0,0 +1,14 @@
+class AddAnnotationModel < ActiveRecord::Migration[6.0]
+  def change
+    create_table "annotations" do |t|
+      t.references :created_by, foreign_key: { to_table: :users }
+      t.references :map_overlay, foreign_key: { to_table: :map_overlays }
+      t.references :building, foreign_key: { to_table: :buildings }
+      t.text "annotation_text"
+    end
+    add_reference :map_overlays, :annotations, index: true
```

Comment:
Remove this! The foreign keys on `annotations` are enough. No need to add a reference on the buildings or map layers - which effectively say "a map layer can only have one annotation period" and the same for a building. Whereas the point of this exercise is to allow one annotation per building/map layer combination.

Reply:
Another migration needs to sweep through the buildings with `annotation_legacy` and create an annotation record to the default map layer.